InternxtDesktop/Views/Widget/WidgetBackup.swift

```diff
@@ -0,0 +1,227 @@
+                        AppText("\(backupStorageValue ?? 0)")
+                            .font(.SMMedium)
+                            .foregroundColor(.Gray60)
+
+                        AppText(backupStorageUnit ?? "")
+                            .font(.XSMedium)
+                            .foregroundColor(.Gray60)
+                    }
+                    .padding([.vertical], 4)
+                    .padding([.horizontal], 8)
+                    .background(Color.Gray5)
+                    .cornerRadius(8)
+                }
+            }
+
+            if isLoading {
+                // show progress bar
+                ProgressBar
+            }
+
+        }
+        .padding(16)
+        .background(colorScheme == .dark ? Color.Gray5 : Color.white)
+        .cornerRadius(8)
+        .shadow(color: .black.opacity(0.05), radius: 1, x: 0, y: 1)
+        .overlay(
+            RoundedRectangle(cornerRadius: 8)
+                .inset(by: -0.5)
+                .stroke(Color.Gray10, lineWidth: 1)
+        )
+    }
+
+    var ProgressField: some View {
+        HStack(spacing: 4) {
+            Image(systemName: "arrow.up.circle.fill")
+                .resizable()
+                .frame(width: 16, height: 16)
+                .foregroundColor(.Primary)
+
+            AppText("BACKUP_BACKING_UP")
+                .font(.SMMedium)
+                .foregroundColor(.Primary)
+        }
+    }
+
+    var ProgressBar: some View {
+        ZStack(alignment: .leading) {
+            GeometryReader { proxy in
+                RoundedRectangle(cornerRadius: 100)
+                    .fill(Color.Gray5)
+                    .frame(maxWidth: .infinity, minHeight: 4, maxHeight: 4)
+                    .onAppear {
+                        progressBarWidth = proxy.size.width
+                    }
+            }
+
+            RoundedRectangle(cornerRadius: 100)
+                .fill(Color.Primary)
+                .frame(width: progressBarWidth * progress, height: 4)
+        }
+        .frame(height: 5, alignment: .leading)
+    }
+
+    var ButtonsToRender: some View {
+        HStack(spacing: 8) {
+            AppButton(title: "BACKUP_STOP_BACKUP", onClick: {
+
+            }, type: .primary)
+
+            AppButton(title: "BACKUP_BROWSE_FILES", onClick: {
+
+            }, type: .secondary)
+        }
+        .frame(maxWidth: .infinity)
+    }
+
+    var FolderSelection: some View {
+        VStack(alignment: .leading, spacing: 8) {
+            AppText("BACKUP_SELECTED_FOLDERS")
+                .font(.SMMedium)
+                .foregroundColor(.Gray80)
+
+            HStack(spacing: 10) {
+                AppButton(title: "BACKUP_CHANGE_FOLDERS", onClick: {
+
+                }, type: .secondary)
+
+                Text("BACKUP_\("\(numOfFolders)")_NUMBER_OF_FOLDERS_SELECTED")
+                    .font(.SMRegular)
+                    .foregroundColor(.Gray60)
+            }
+        }
+    }
+
+    var UploadFrequency: some View {
```

Comment:
views and files are free, meaning that you can create multiple views, with reduced logic, don't bundle everything into a single view, or you will end up with pretty large view files, that are hard to read, and the logic becomes complex